test/functional/tests/volumes/test_many_cores_on_many_lvms.py

```diff
@@ -0,0 +1,117 @@
+#
+# Copyright(c) 2021 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+import datetime
+
+import pytest
+
+from api.cas import casadm
+from api.cas.init_config import InitConfig
+from core.test_run import TestRun
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from storage_devices.lvm import Lvm, LvmConfiguration, get_block_devices_list
+from test_tools import fs_utils
+from test_tools.fio.fio import Fio
+from test_tools.fio.fio_param import IoEngine, ReadWrite, VerifyMethod
+from test_utils.size import Size, Unit
+
+opencas_conf_path = "/etc/opencas/opencas.conf"
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core1", DiskTypeLowerThan("cache"))
+def test_many_cores_on_many_lvms():
+    """
+        title: Test for CAS creation with lvms as cores: 1 cache, 16 lvms, 16 cores.
+        description: |
+          Validation of LVM support, CAS with 1 cache and 16 lvms as 16 cores.
+        pass_criteria:
+          - LVMs created successfully.
+          - CAS devices created successfully.
+          - FIO with verification ran successfully.
+          - Configuration after reboot match configuration before.
+    """
+    with TestRun.step(f"Prepare devices."):
+
+        cache_dev, core_dev = prepare_devices()
+
+    with TestRun.step("Create LVMs."):
+        lvm_filters = []
+        pv_num = 1
+        vg_num = 1
+        lv_num = 16
+        cache_num = 1
+        cas_dev_num = 16
+
+        config = LvmConfiguration(lvm_filters, pv_num, vg_num, lv_num, cache_num, cas_dev_num)
+
+        lvms = Lvm.create_specific_lvm_configuration(core_dev, config, lvm_as_core=True)
+
+    with TestRun.step(f"Create CAS device."):
+        cache = casadm.start_cache(cache_dev[0], force=True)
+        cores = []
+        for lvm in lvms:
+            cores.append(cache.add_core(lvm))
+
+    with TestRun.step("Run FIO with verification on LVM."):
+        run_fio_on_lvm(cores)
+
+    with TestRun.step("Flush buffers"):
+        for core in cores:
+            TestRun.executor.run(f"hdparm -f {core.path}")
+
+    with TestRun.step("Create init config from running configuration"):
+        InitConfig.create_init_config_from_running_configuration()
+        config_before_reboot = fs_utils.read_file(opencas_conf_path)
+        devices_before = get_block_devices_list()
```

Comment:
I mentioned below about creating file with some common methods - this can also be extracted together with `opencas_conf_path` - I saw this step in your tests from this category.